packages/components/src/components/amount-input/amount-input.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { isMobile } from '@deriv/shared';
+import Input from '../input';
+import Text from '../text';
+
+type TAmountInput = {
+    currency: string;
+    decimal_places?: number;
+    disabled?: boolean;
+    initial_value?: number;
+    label?: string;
+    locale?: Intl.LocalesArgument;
+    max_digits?: number;
+    onChange?: (value: number) => void;
+};
+
+const AmountInput = ({
+    currency,
+    decimal_places = 2,
+    disabled = false,
+    initial_value = 0,
+    label,
+    locale,
+    max_digits = 8,
+    onChange,
+}: TAmountInput) => {
+    const [value, setValue] = useState(initial_value);
+    const [focus, setFocus] = useState(false);
+    const [is_pasting, setIsPasting] = useState(false);
+    const [caret_right_offset, setCaretRightOffset] = useState(0);
+    const [selection, setSelection] = useState<{
+        selectionStart: number;
+        selectionEnd: number;
+    }>({ selectionStart: 0, selectionEnd: 0 });
+    const [target, setTarget] = useState<React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>['target']>();
+
+    const displayNumber = useCallback(
+        (number: number) => number.toLocaleString(locale, { minimumFractionDigits: decimal_places }),
+        [decimal_places, locale]
+    );
+
+    useEffect(() => {
+        // update caret position every time the value changes (this happens after onChange)
+        const updated_caret_position = displayNumber(value).length - caret_right_offset;
+        target?.setSelectionRange(updated_caret_position, updated_caret_position);
+        setSelection({ selectionStart: updated_caret_position, selectionEnd: updated_caret_position });
+    }, [value, target, displayNumber]);
```

Comment:
Not adding `caret_right_offset` to deps because it causes wrong behaviour for caret when using arrow keys